packages/plugins/plugin-sketch/src/types/diagram.ts

```diff
@@ -17,6 +18,8 @@ export class CanvasType extends TypedObject({ typename: 'dxos.org/type/Canvas',
 export class DiagramType extends TypedObject({ typename: 'dxos.org/type/Diagram', version: '0.1.0' })({
   name: S.optional(S.String),
   canvas: ref(CanvasType),
+  // Associated threads.
+  threads: S.optional(S.mutable(S.Array(ref(ThreadType)))),
```

Comment:
when does this array get initialized?

Reply:
https://github.com/dxos/dxos/blob/e79c8413c95fb86c9da42584406217085a289d9c/packages/plugins/plugin-thread/src/ThreadPlugin.tsx#L318-L327

Here currently.